app/containers/MessageBox/RecordAudio.js

```diff
@@ -0,0 +1,201 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { View, Text } from 'react-native';
+import { activateKeepAwake, deactivateKeepAwake } from 'expo-keep-awake';
+import { Audio } from 'expo-av';
+import { getInfoAsync } from 'expo-file-system';
+import { BorderlessButton } from 'react-native-gesture-handler';
+
+import { CustomIcon } from '../../lib/Icons';
+import styles from './styles';
+import I18n from '../../i18n';
+import { themes } from '../../constants/colors';
+
+const RECORDING_EXTENSION = '.aac';
+const RECORDING_SETTINGS = {
+	android: {
+		extension: RECORDING_EXTENSION,
+		outputFormat: Audio.RECORDING_OPTION_ANDROID_OUTPUT_FORMAT_AAC_ADTS,
+		audioEncoder: Audio.RECORDING_OPTION_ANDROID_AUDIO_ENCODER_AAC,
+		sampleRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.android.sampleRate,
+		numberOfChannels: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.android.numberOfChannels,
+		bitRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.android.bitRate
+	},
+	ios: {
+		extension: RECORDING_EXTENSION,
+		audioQuality: Audio.RECORDING_OPTION_IOS_AUDIO_QUALITY_MIN,
+		sampleRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.sampleRate,
+		numberOfChannels: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.numberOfChannels,
+		bitRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.bitRate,
+		outputFormat: Audio.RECORDING_OPTION_IOS_OUTPUT_FORMAT_MPEG4AAC
+	}
+};
+const RECORDING_MODE = {
+	allowsRecordingIOS: true,
+	playsInSilentModeIOS: true,
+	staysActiveInBackground: false,
+	shouldDuckAndroid: true,
+	playThroughEarpieceAndroid: false,
+	interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
+	interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX
+};
+
+const _formatTime = function(seconds) {
+	let minutes = Math.floor(seconds / 60);
+	seconds %= 60;
+	if (minutes < 10) { minutes = `0${ minutes }`; }
+	if (seconds < 10) { seconds = `0${ seconds }`; }
+	return `${ minutes }:${ seconds }`;
+};
+
+const startRecordingAudio = async(instance, setRecordingStatus, setRecorderBusy) => {
```

Comment:
I get it you're trying to make abstractions here, but think on the next dev who is going to do maintenance here.
`startRecordingAudio` receives an `instance` of what?
I have to read this whole file in order to be sure it needs to receive an instance of `new Audio.Recording()`.
Why not making this whole code a React.Component and make it a class function?

Reply:
Wouldn't renaming the variable or moving the function into the component be a better option? Is it really necessary to make it a class component as we're not using lifecycle hooks which I think could be one of the main reasons to make the switch

Reply:
That's not the only reason to use class components.
It'd make your code much more readable if those function could access a class variable: `this.recording = new Audio.Recording()`

This way you wouldn't be relying on passing instances as params to functions.